core/query/index.go

```diff
@@ -108,17 +108,16 @@ func (ind *Indexer) insertAnnotatedOutputs(ctx context.Context, b *bc.Block, ann
 		outputTxPositions pg.Uint32s
 		outputIndexes     pg.Uint32s
 		outputTxHashes    pq.ByteaArray
+		outputIDs         pq.ByteaArray
 		outputData        pq.StringArray
-		prevoutHashes     pq.ByteaArray
-		prevoutIndexes    pg.Uint32s
+		prevoutOIDs       pq.ByteaArray
 	)
 
 	for pos, tx := range b.Transactions {
 		for _, in := range tx.Inputs {
 			if !in.IsIssuance() {
-				outpoint := in.Outpoint()
-				prevoutHashes = append(prevoutHashes, outpoint.Hash[:])
-				prevoutIndexes = append(prevoutIndexes, outpoint.Index)
+				oid := in.OutputID()
+				prevoutOIDs = append(prevoutOIDs, oid[:])
```

Comment:
nit: `OID` might get confused with Postgres's [object identifiers](https://www.postgresql.org/docs/8.1/static/datatype-oid.html). Maybe rename to `prevoutIDs`?